shared/src/commonMain/kotlin/co/nimblehq/ic/kmm/suv/di/module/RemoteModule.kt

```diff
@@ -2,11 +2,14 @@ package co.nimblehq.ic.kmm.suv.di.module
 
 import co.nimblehq.ic.kmm.suv.data.remote.datasource.TokenRemoteDataSource
 import co.nimblehq.ic.kmm.suv.data.remote.datasource.TokenRemoteDataSourceImpl
-import co.nimblehq.ic.kmm.suv.data.remote.httpclient.core.provideHttpClient
-import org.koin.core.module.dsl.singleOf
+import co.nimblehq.ic.kmm.suv.data.remote.apiclient.core.ApiClient
+import org.koin.core.qualifier.named
 import org.koin.dsl.module
 
+private const val DEFAULT_HTTP_CLIENT = "default"
+
 val remoteModule = module {
-    singleOf(::provideHttpClient)
-    single<TokenRemoteDataSource> { TokenRemoteDataSourceImpl(get()) }
+    single { ApiClient(get(), get()) }
+    single(named(DEFAULT_HTTP_CLIENT)) { ApiClient(get()) }
+    single<TokenRemoteDataSource> { TokenRemoteDataSourceImpl(get(named(DEFAULT_HTTP_CLIENT))) }
```

Comment:
Do you mean API_CLIENT here also?

```suggestion
    single(named(DEFAULT_API_CLIENT)) { ApiClient(get()) }
    single<TokenRemoteDataSource> { TokenRemoteDataSourceImpl(get(named(DEFAULT_API_CLIENT))) }
```

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/94/commits/fca5d8acb0b263ec50661cd79bd5560e4cad0f8f

I changed to `UNAUTHORIZED_API_CLIENT` for easy to understand